maven-core/src/main/java/org/apache/maven/project/ReactorModelPool.java

```diff
@@ -48,13 +49,20 @@ class ReactorModelPool {
      * @throws IllegalStateException if version was null and multiple modules share the same groupId + artifactId
      */
     public Model get(String groupId, String artifactId, String version) {
-        return modelsByGa.getOrDefault(new GAKey(groupId, artifactId), Collections.emptySet()).stream()
-                .filter(m -> version == null || version.equals(getVersion(m)))
-                .reduce((a, b) -> {
-                    throw new IllegalStateException(
-                            "Multiple modules with key " + a.getGroupId() + ':' + a.getArtifactId());
-                })
-                .orElse(null);
+        Collection<Model> models = modelsByGa.getOrDefault(new GAKey(groupId, artifactId), Collections.emptySet());
+        if (models.isEmpty()) {
+            return null;
+        } else if (models.size() == 1) {
+            return models.iterator().next();
+        } else {
+            return models.stream()
+                    .filter(m -> version == null || version.equals(getVersion(m)))
+                    .reduce((a, b) -> {
+                        throw new IllegalStateException(
+                                "Multiple modules with key " + a.getGroupId() + ':' + a.getArtifactId());
+                    })
+                    .orElse(null);
+        }
```

Comment:
This does not change the behaviour, right ? What's the point ?

Reply:
Oh, it does change the behaviour, as if there's a single version, it will always be used, even if there's a version mismatch.  I think that's wrong.
Multiple versions of the same GA inside a reactor is a supported (albeit rare) use case.